firebase-firestore/src/main/java/com/google/firebase/firestore/remote/RemoteStore.java

```diff
@@ -407,8 +416,20 @@ private void startWatchStream() {
         "startWatchStream() called when shouldStartWatchStream() is false.");
     watchChangeAggregator = new WatchChangeAggregator(this);
     watchStream.start();
-
     onlineStateTracker.handleWatchStreamStart();
+
+    DelayedTask connectivityAttemptTimer =
+        workerQueue.enqueueAfterDelay(
+            TimerId.ONLINE_STATE_TIMEOUT,
+            CONNECTIVITY_ATTEMPT_TIMEOUT_MS,
+            () -> {
+              // If the network has been explicitly disabled, make sure we don't accidentally
+              // re-enable it.
+              if (canUseNetwork()) {
```

Comment:
Previously, in the case of an online state timeout we performed the following things:

```java
                logClientOfflineWarningIfNecessary(	
                    String.format(	
                        Locale.ENGLISH,	
                        "Backend didn't respond within %d seconds\n",	
                        ONLINE_STATE_TIMEOUT_MS / 1000));	
                setAndBroadcastState(OnlineState.OFFLINE);	
```

This isn't calling `handleWatchStreamFailure` so how is the OnlineState supposed to get to OFFLINE after the first timeout?